msteams-platform/toolkit/add-single-sign-on.md

```diff
@@ -125,236 +127,295 @@ SSO command handler `ProfileSsoCommandHandler` uses an Azure AD token to call Mi
+     await commandBot.requestHandler(req, res);
+   });
+   ```
+
+   with
+
+   ```ts
+   server.post("/api/messages", async (req, res) => {
+     await commandBot.requestHandler(req, res).catch((err) => {
+       // Error message including "412" means it is waiting for user's consent, which is a normal process of SSO, sholdn't throw this error.
+       if (!err.message.includes("412")) {
+         throw err;
+       }
+     });
+   });
+   ```
 
 1. Replace the options for `ConversationBot` instance in `bot/src/internal/initialize` to add the SSO config and SSO command handler:
 
-    ```ts
-    export const commandBot = new ConversationBot({
-        ...
-        command: {
-            enabled: true,
-            commands: [new HelloWorldCommandHandler()],
-        },
-    });
-    ```
-
-    with
-
-    ```ts
-    import { ProfileSsoCommandHandler } from "../profileSsoCommandHandler";
-
-    export const commandBot = new ConversationBot({
-        ...
-        // To learn more about ssoConfig, please refer teamsfx sdk document: https://docs.microsoft.com/microsoftteams/platform/toolkit/teamsfx-sdk
-        ssoConfig: {
-            aad :{
-                scopes:["User.Read"],
-            },
-        },
-        command: {
-            enabled: true,
-            commands: [new HelloWorldCommandHandler() ],
-            ssoCommands: [new ProfileSsoCommandHandler()],
-        },
-    });
-    ```
+   ```ts
+   export const commandBot = new ConversationBot({
+       ...
+       command: {
+           enabled: true,
+           commands: [new HelloWorldCommandHandler()],
+       },
+   });
+   ```
+
+   with
+
+   ```ts
+   import { ProfileSsoCommandHandler } from "../profileSsoCommandHandler";
+
+   export const commandBot = new ConversationBot({
+       ...
+       // To learn more about ssoConfig, please refer teamsfx sdk document: https://docs.microsoft.com/microsoftteams/platform/toolkit/teamsfx-sdk
+       ssoConfig: {
+           aad :{
+               scopes:["User.Read"],
+           },
+       },
+       command: {
+           enabled: true,
+           commands: [new HelloWorldCommandHandler() ],
+           ssoCommands: [new ProfileSsoCommandHandler()],
+       },
+   });
+   ```
 
 1. Register your command in the Teams app manifest. Open `templates/appPackage/manifest.template.json`, and add following lines under `commands` in `commandLists` of your bot:
 
-    ```json
-    {
-        "title": "profile",
-        "description": "Show user profile using Single Sign On feature"
-    }
-    ```
+   ```json
+   {
+     "title": "profile",
+     "description": "Show user profile using Single Sign On feature"
+   }
+   ```
 
 #### Add a new SSO command to the bot (Optional)
 
 After successfully adding SSO in your project, you can add a new SSO command.
 
 1. Create a new file such as `photoSsoCommandHandler.ts` or `photoSsoCommandHandler.js` in `bot/src/` and add your own SSO command handler to call Graph API:
```

Comment:
Is the code after this sentence related to this line. If yes, add a line about the code. Something in the lines of the following is an example of.... sample code.